crates/sui-framework/src/natives/object_runtime/object_store.rs

```diff
@@ -81,6 +81,24 @@ impl<'a> Inner<'a> {
                     PartialVMError::new(StatusCode::STORAGE_ERROR).with_message(format!("{msg}"))
                 })?;
             let obj_opt = if let Some(object) = child_opt {
+                // guard against bugs in `read_child_object`: if it returns a child object such that
+                // C.parent != parent, we raise an invariant violation
+                match &object.owner {
+                    Owner::ObjectOwner(id) => {
+                        if ObjectID::from(*id) != parent {
+                            return Err(PartialVMError::new(StatusCode::STORAGE_ERROR).with_message(
```

Comment:
STORAGE_ERROR is an invariant violation I believe, right?

Reply:
Yes

Reply:
It feels like the right one! Just wanted to double check (you would think I would remember one way or another since I wrote this other code that uses it...)